### ISSUE TITLE
Android: Overriding default actions

### ISSUE THREAD

**Issue by amarlette** [facebook/react-native#30855]
## Description
Using the `accessibilityActions` prop you can override the default usage hints for the actions "activate" and "longpress" on Android by giving them a new label.  While this works on most components, on `<Button>`, `<Text>`, `<TextInput>`, and `<Picker>` this has no effect.

## React Native version:
v0.63

## Expected Behavior
If the label for "activate" or "longpress" is overridden via the `accessibilityActions` prop, screen readers should announce the hint of "Double-tap to `<your custom label>`" or "Double-tap and hold to `<your custom label>`" rather than "Double-tap to activate" or "Double-tap and hold to longpress".

## Android Details
Since this properly works on most components, this is likely an issue on the Javascript side of these props simply not being passed along.

**Comment by kacieb** [comment]
> While this works on most components, on , , , and this has no effect.

Which components is this needed for? Are there separate issues for these? cc @blavalla @amarlette

**Comment by blavalla** [comment]
Oops, my component names without markdown formatting were being interpreted as HTML that github was trying to render!  I updated the description to call out the specific components with issues.

**Comment by huzaifaaak** [comment]
@blavalla I was going though the documentation for [accessibilityActions](https://reactnative.dev/docs/accessibility#accessibility-actions) and I realized that we have to implement `onAccessibilityAction` aswell. Can you shed some light on how the custom accessibilityActions works?
How can I reproduce custom actions?

**Comment by blavalla** [comment]
@huzaifaaak, on Android, React Native's "accessibilityActions" API (which would include the accessibilityActions and onAccessibilityAction props) has three main purposes:

1.) To override the default "usage hint" label for the click or long-click actions, which is something Talkback allows you to do.
2.) To create a custom accessibility action, which can be triggered via Talkback's actions menu.
3.) To change the behavior that occurs when a specific accessibility action (either custom, or a built-in one) is triggered.

To achieve #1, you'd only need to use the accessibilityActions prop, and add a label for either the "activate" or "longpress" actions.  Setting a label here would change Talkback from saying "Double-tap to activate" to "Double-tap to <your custom label>".  This feature is what this issue is describing, and what currently doesn't work on all components.

To create a fully custom action (#2), you'd need to give the action a name, a label, and then use the onAccessibilityAction prop to define behavior for that action. I am not sure if that is currently working for the listed problematic components, but my guess would be that it's not, as I assume the entire `accessibilityActions` prop is simply not supported.  

To support #3, I think you'd *only* need to implement the `onAccessibilityAction` prop, but you'd only be able to use it to override built-in actions such as activate, longpress on Android, or magicTap on iOS.

Feel free to investigate a bit to determine if use cased 2 and 3 are working not, I think a fix could be made at the same time as the fix to support overriding the usage hints, as they are likely connected problems.  Let me know if you have any other questions!

**Comment by fabriziobertoglio1987** [comment]
This is the result of my quick research on #1

**accessibilityActions** prop

Javascript
https://github.com/facebook/react-native/blob/83e5cdd369de637bbf9ea20d93cebc9984efaabb/Libraries/Components/View/ViewPropTypes.js#L433

https://github.com/facebook/react-native/blob/83e5cdd369de637bbf9ea20d93cebc9984efaabb/Libraries/Components/View/ViewPropTypes.js#L46

Java
https://github.com/facebook/react-native/blob/83e5cdd369de637bbf9ea20d93cebc9984efaabb/ReactAndroid/src/main/java/com/facebook/react/uimanager/BaseViewManager.java#L259

https://github.com/facebook/react-native/blob/83e5cdd369de637bbf9ea20d93cebc9984efaabb/ReactAndroid/src/main/java/com/facebook/react/uimanager/ReactAccessibilityDelegate.java#L205

The accessibilityActions seems to be mapped on the Java side depending on the type of components and the actions available. This actions should be available to all components that implement ReactAccessibilityDelegate.

https://github.com/facebook/react-native/blob/83e5cdd369de637bbf9ea20d93cebc9984efaabb/ReactAndroid/src/main/java/com/facebook/react/uimanager/ReactAccessibilityDelegate.java#L54-L59

for example in the case of the Slider, the ReactSliderManager includes the following type of actions

https://github.com/facebook/react-native/blob/83e5cdd369de637bbf9ea20d93cebc9984efaabb/ReactAndroid/src/main/java/com/facebook/react/views/slider/ReactSliderManager.java#L281-L286

The callback will trigger the following methods mapped in ReactSliderManager

https://github.com/facebook/react-native/blob/83e5cdd369de637bbf9ea20d93cebc9984efaabb/ReactAndroid/src/main/java/com/facebook/react/views/slider/ReactSliderManager.java#L289-L298

the method triggers the following announcement

https://github.com/facebook/react-native/blob/83e5cdd369de637bbf9ea20d93cebc9984efaabb/ReactAndroid/src/main/java/com/facebook/react/views/slider/ReactSliderManager.java#L94-L107

This is what I found for **onAccessibilityAction** callback in java

https://github.com/facebook/react-native/blob/83e5cdd369de637bbf9ea20d93cebc9984efaabb/ReactAndroid/src/main/java/com/facebook/react/uimanager/BaseViewManager.java#L431-L435

I'll keep researching. Thanks :pray:

**Comment by huzaifaaak** [comment]
@amarlette #31532 fixes this issue.
